Simplified/Reader2/Networking/NYPLAnnotations.swift

```diff
@@ -595,16 +594,17 @@ final class NYPLAnnotations: NSObject, NYPLAnnotationSyncing {
     return annotationID
   }
 
-  /// Annotation-syncing is possible only if the given `account` is signed-in
+  /// Annotation-syncing is possible only if the current user account is signed-in
   /// and if the currently selected library supports it.
-  class func syncIsPossible(_ account: NYPLUserAccount) -> Bool {
+  private class func syncIsPossible() -> Bool {
     let library = AccountsManager.shared.currentAccount
-    return account.hasCredentials() && library?.details?.supportsSimplyESync == true
+    let userAccount = NYPLUserAccount.sharedAccount()
```

Comment:
@jkim-git  I made NYPLUserAccount implicit because if we go the hybrid route the need for the user account status will be entirely handled by the web side. Essentially it's an implementation detail. 

In a fully native world of course having this as an implicit dependency would not be good.